src/ipred_prepare.rs

```diff
@@ -149,7 +149,7 @@ pub unsafe fn rav1d_prepare_intra_edges<BD: BitDepth>(
         0
     };
     let dst = if last_pos > first_pos {
-        slice::from_raw_parts(dst.offset(first_pos), (last_pos - first_pos) as usize)
+        unsafe { slice::from_raw_parts(dst.offset(first_pos), (last_pos - first_pos) as usize) }
```

Comment:
You can't do this yet and mark the `fn` safe, since to be safe, since calling a safe `fn` must never cause UB.  If it does, then it's unsound.  We can't guarantee that the `dst` pointer is valid in this way, so we have to leave the `fn` as `unsafe`.  And generally, when you add an `unsafe` block like this, you want to add a `// Safety: reason` comment explaining why this is safe, i.e. why this can never be UB, no matter how the `fn` is called.  We're very close, though; once `dst` is made into a slice (or potentially some other type that safely encapsulates the raw pointer), we can mark this `fn` safe.